x/valset-pref/validator_set_test.go

```diff
@@ -274,3 +205,132 @@ func (s *KeeperTestSuite) TestIsPreferenceValid() {
 		})
 	}
 }
+
+// NOTE: this is the case that used to error. Fixed by this PR
```

Comment:
We def need more test cases here!

Reply:
added extensive test including the error test in tdd format: 
https://github.com/osmosis-labs/osmosis/pull/5967/files#diff-e8d94c443b26e1cda354d150b30ce5542344099e0bbbb8e84e507d9df791daf7R393-R461